python/tvm/driver/tvmc/autotuner.py

```diff
@@ -0,0 +1,301 @@
+    """Get the tuning tasks for a given relay module.
+
+    Parameters
+    ----------
+    mod : tvm.relay.Module
+        The relay module from which to extract tuning tasks.
+    params : dict
+        The params for the relay module.
+    target : tvm.target.Target
+        The compilation target.
+    target_host : str, optional
+        The compilation target for the host.
+    alter_layout : str, optional
+        The layout to convert the graph to. Note, the convert layout
+        pass doesn't currently guarantee the whole of the graph will
+        be converted to the chosen layout.
+
+    Returns
+    -------
+    tasks : list of autotvm.Tasks
+        list of tasks to be tuned
+
+    """
+    if not target_host:
+        target_host = target
+
+    if alter_layout:
+        mod = common.convert_graph_layout(mod, alter_layout)
+
+    tasks = autotvm.task.extract_from_program(
+        mod["main"],
+        target=target,
+        target_host=target_host,
+        params=params,
+        ops=(relay.op.get("nn.conv2d"),),
+    )
+
+    return tasks
+
+
+def tune_tasks(
+    tasks,
+    log_file,
+    measure_option,
+    tuner,
+    trials,
+    early_stopping=None,
+    tuning_records=None,
+):
+    """Tune a list of tasks and output the history to a log file.
+
+    Parameters
+    ----------
+    tasks : list
+        A list of autotvm.Tasks to tune.
+    log_file : str
+        A file to output the tuning history, in JSON.
+    measure_option : autotvm.measure_option
+        Options to build and run a tuning task.
+    tuner : str
+        Which tuner to use.
+    trials : int
+        The maximum number of tuning trials to perform.
+    early_stopping : int, optional
+        The minimum number of tuning trials to perform.
+        This will be equal to 'trials' if not specified.
+    tuning_records: str, optional
+        Path to the file produced by the tuning, to be used during
+        tuning.
+
+    """
+    if not tasks:
+        logging.warning("there were no tasks found to be tuned")
+        return
+
+    if not early_stopping:
+        early_stopping = trials
+
+    for i, tsk in enumerate(tasks):
+        prefix = "[Task %2d/%2d] " % (i + 1, len(tasks))
+
+        # Create a tuner
+        if tuner in ("xgb", "xgb-rank"):
+            tuner_obj = XGBTuner(tsk, loss_type="rank")
+        elif tuner == "xgb_knob":
+            tuner_obj = XGBTuner(tsk, loss_type="rank", feature_type="knob")
+        elif tuner == "ga":
+            tuner_obj = GATuner(tsk, pop_size=50)
+        elif tuner == "random":
+            tuner_obj = RandomTuner(tsk)
+        elif tuner == "gridsearch":
+            tuner_obj = GridSearchTuner(tsk)
+        else:
+            raise TVMCException("invalid tuner: %s " % tuner)
+
+        # If transfer learning is being used, load the existing results
+        if tuning_records and os.path.exists(tuning_records):
+            logging.warning("loading tuning records from %s", tuning_records)
```

Comment:
`info` should be a proper logging level here.

Reply:
Yep. Done